_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByName, validateSnapshotTimestamp } from './helpers/helpers';
+
+test.describe('Snapshot Repositories', () => {
+  test('Snapshot a repository', async ({ page }) => {
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+
+    const repoName = 'one';
+
+    await test.step('Cleanup repository, if using the same url', async () => {
+      await deleteAllRepos(
+        page,
+        `&url=https://jlsherrill.fedorapeople.org/fake-repos/revision/` + repoName,
+      );
+    });
+
+    await test.step('Open the add repository modal', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByText('Add custom repositories')).toBeVisible();
+    });
+
+    await test.step('Select snapshotting', async () => {
+      await page.getByRole('radio', { name: 'Snapshotting' }).first().click();
+    });
+
+    await test.step('Fill in the repository details', async () => {
+      await page.getByLabel('Name').fill(repoName);
+      await page
+        .getByLabel('URL')
+        .fill('https://jlsherrill.fedorapeople.org/fake-repos/revision/' + repoName);
+    });
+
+    await test.step('Filter by architecture', async () => {
+      await page.getByRole('button', { name: 'filter architecture' }).click();
+      await page.getByRole('option', { name: 'x86_64' }).click();
+    });
+
+    await test.step('Filter by version', async () => {
+      const versionFilterButton = page.getByRole('button', { name: 'filter version' });
+      await versionFilterButton.click();
+      await page.getByRole('menuitem', { name: 'el9' }).locator('label').click();
+      await page.getByRole('menuitem', { name: 'el8' }).locator('label').click();
+      await versionFilterButton.click(); // Close the version filter dropdown
+    });
+
+    await test.step('Wait for the successful API call', async () => {
+      if (await page.locator('.pf-v5-c-helper-text__item.pf-m-error').isVisible()) {
```

Comment:
This is racy by design. This might pass just fine, because when this is being checked there might not be any error yet. If you want check like this, you want to add listener, similar to https://github.com/content-services/content-sources-frontend/blob/main/_playwright-tests/UI/helpers/helpers.ts#L12